pkg/util/docker/dockerfile/dockerfile.go

```diff
@@ -0,0 +1,55 @@
+package dockerfile
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/docker/docker/builder/parser"
+)
+
+// ParseTreeToDockerfile takes a Dockerfile AST node, as generated by
+// parser.Parse, and returns a new, equivalent, Dockerfile.
+func ParseTreeToDockerfile(node *parser.Node) []byte {
+	buf := []byte(node.Original)
+	for _, child := range node.Children {
+		buf = append(buf, ParseTreeToDockerfile(child)...)
+	}
+	// Append a line break when needed.
+	if len(buf) > 0 && buf[len(buf)-1] != '\n' {
+		buf = append(buf, '\n')
+	}
+	return buf
+}
+
+// FindAll returns the indices of all children of node such that
+// node.Children[i].Value == command.
+func FindAll(node *parser.Node, command string) []int {
```

Comment:
`s := FindAll(node, command.Cmd)` would find all occurrences, then you can decide to consider only the last one: `s[len(s)-1]`. In this case, your requirement would be better expressed as `FindLast(node *parser.Node, cmd string) int`. The problem with returning an int is that you want to be able to tell the caller that cmd was not found... I'm having just `FindAll` for now.